presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoMetadata.java

```diff
@@ -301,4 +312,39 @@ private static List<MongoColumnHandle> buildColumnHandles(ConnectorTableMetadata
                 .map(m -> new MongoColumnHandle(m.getName(), m.getType(), m.isHidden()))
                 .collect(toList());
     }
+
+    @Override
+    public ColumnHandle getDeleteRowIdColumnHandle(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return new MongoColumnHandle(DELETE_ROW_ID, BIGINT, true);
+    }
+
+    @Override
+    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return tableHandle;
+    }
+
+    @Override
+    public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHandle, Collection<Slice> fragments)
+    {
+        MongoTableHandle table = (MongoTableHandle) tableHandle;
+        OptionalLong.of(mongoSession.deleteDocuments(table.getSchemaTableName(), Optional.ofNullable(constraint.getSummary())));
```

Comment:
Wouldn't getSummary() throw an NPE if constraint is not set in line 183?

Reply:
I have handled the NPE and added the commit, Could you please check.